### PR TITLE
Repository refactor/restructure/cleanup

### PR DESCRIPTION
**Pull request by TriplEight**
- restructured dirs, now all CI jobs match the image name and dockerfile's dir
- added metadata to all images we have
- optimized CI
- removed stale dockerfiles
- optimized layers in build dockerfiles
- updated `README.md`s
- added caching for snap builds
- some cleanup
- fixed `armhf`, `arm64`, `i386` as they didn't inherit from the modern `parity-ci-linux` image

Please tell me before merging, https://github.com/paritytech/parity-ethereum/pull/10863 should be merged right after this.

### CODE REVIEW

File: docker-files-for-Gitlab-CI-rust/cross/i686/Dockerfile [hunk 1]
```diff
@@ -8,26 +8,36 @@ FROM parity/rust-parity-ethereum-build:xenial as builder
...
+	CXX=g++ \
+	CARGO_TARGET=i686-unknown-linux-gnu \
+	HOST_CC=gcc \
+	HOST_CXX=g++ \
+	LDFLAGS="-L/usr/lib32" \
+	CPPFLAGS="-I/usr/include" \
+# cache handler
+	RUSTC_WRAPPER=sccache \
+	CARGO_HOME=/ci-cache/parity-ethereum/cargo \
+	SCCACHE_DIR=/ci-cache/parity-ethereum/sccache \
```
Comment (gabreal) [inline docker-files-for-Gitlab-CI-rust/cross/i686/Dockerfile]: does the caching respect the architecture?

Comment (TriplEight) [reply docker-files-for-Gitlab-CI-rust/cross/i686/Dockerfile]: These changes are waiting for a new config file in `parity-ethereum/scripts/gitlab/` which will have our current target-linker pairs. 
I'll test before merging.
These parameters respect the setup of caching in all our docker builder images.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -42,156 +42,153 @@ variables:
...
-    CONTAINER_IMAGE:     $CI_JOB_NAME
-    CONTAINER_TAG:       latest
-    DOCKERFILE_DIR:      docker/helpers/tools
+    CONTAINER_IMAGE:               $CI_JOB_NAME
+    CONTAINER_TAG:                 latest
+    DOCKERFILE_DIR:                $CI_JOB_NAME
   only:
     variables:
       - $DOCKERIMAGE == $CI_JOB_NAME
       - $DOCKERIMAGE == "tools"
```
Comment (gabreal) [inline .gitlab-ci.yml]: I guess this line could be also removed as with the other images.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -42,156 +42,153 @@ variables:
...
-    CONTAINER_IMAGE:     $CI_JOB_NAME
-    CONTAINER_TAG:       latest
-    DOCKERFILE_DIR:      docker/helpers/awscli
+    CONTAINER_IMAGE:               $CI_JOB_NAME
+    CONTAINER_TAG:                 latest
+    DOCKERFILE_DIR:                $CI_JOB_NAME
   only:
     variables:
       - $DOCKERIMAGE == $CI_JOB_NAME
       - $DOCKERIMAGE == "awscli"
```
Comment (gabreal) [inline .gitlab-ci.yml]: also this one can potentially be removed, see below.

**Pull request merged**